src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -759,5 +785,75 @@ private static bool IsPotentialFilterCharacter(char c)
                 || char.IsNumber(c)
                 || c == '_';
         }
+
+        private static int GetAggressiveDefaultsMatch(List<MatchResult<VSCompletionItem>> itemsWithMatch, ImmutableArray<string> defaults)
+        {
+            Debug.Assert(!defaults.IsDefaultOrEmpty);
+
+            foreach (var defaultText in defaults)
+            {
+                for (var i = 0; (i < itemsWithMatch.Count); ++i)
+                {
+                    var itemWithMatch = itemsWithMatch[i];
+                    if (itemWithMatch.RoslynCompletionItem.FilterText == defaultText)
+                    {
+                        if (itemWithMatch.PatternMatch == null || itemWithMatch.PatternMatch.Value.Kind <= PatternMatchKind.Prefix)
+                            return i;
```

Comment:
the lack of any sort of commenting/explanations here is really not good.  completion is one of hte most compliated beasts we have, and this is adding a lot of special logic to it.  we really need to make it as clean and crisp as possible, with clear explanations in code as to the *why* of what's going on.

ideally this should be all be extracted out into a super clean helper type that has clear responsibilities and does each part for cleanly (and potentially in a testable fashion).